hikari/events/message_events.py

```diff
@@ -136,7 +136,7 @@ def channel(self) -> typing.Optional[channels.GuildTextChannel]:
         return typing.cast("channels.GuildTextChannel", self.app.cache.get_guild_channel(self.channel_id))
 
     @property
-    def guild(self) -> typing.Optional[guilds.GatewayGuild]:
+    def available_guild(self) -> typing.Optional[guilds.GatewayGuild]:
```

Comment:
Yeah, I think we should revert this change in particular, as it is going to cause a headache for people using this already for not much additional benefit. We should just be able to store a flag on the guild to say if it is available or not.